### PR TITLE
several fixes for reported issues

### PR DESCRIPTION
**Pull request by jonas0b1011001**
Ignore unsupported 'url' properties
More robust targetCalendar selection
Handle RDates when onlyFutureEvents = true
Improved Recurrence Exception Handling 
Add Backoff to cleanup procedure
Add support for Events without uid

### CODE REVIEW

File: Helpers.gs [hunk 1]
```diff
@@ -502,43 +527,41 @@ function checkSkipEvent(event, icalEvent){
...
+      Logger.log("Updating existing event instance");
+      Calendar.Events.update(recEvent, targetCalendarId, eventInstanceToPatch[0].id);
     }
     catch(error){
       Logger.log(error); 
     }
   }
   else{
+    Logger.log("No Instance matched, adding as new event!");
     try{
```
Comment (derekantrican) [inline Helpers.gs:563]: Don't know if we also want to replace this and the `try` above (for `Events.update`) (and maybe more) with `callWithBackoff` like you did below

File: Helpers.gs [hunk 2]
```diff
@@ -108,7 +108,8 @@ function fetchSourceCalendars(sourceCalendarURLs){
  */
 function setupTargetCalendar(targetCalendarName){
   var targetCalendar = Calendar.CalendarList.list().items.filter(function(cal) {
-    return cal.summary == targetCalendarName;
+    var name = cal.summaryOverride || cal.summary;
+    return name == targetCalendarName && cal.accessRole == "owner";
```
[the remark this answers is not in the document]
Comment (jonas0b1011001) [reply Helpers.gs:112]: True, _writer_ should work too. _freeBusyReader_ and _reader_ should be ignored.

File: Helpers.gs (hunk 1 above)
Comment (jonas0b1011001) [reply Helpers.gs:563]: I guess we can add it to all api calls.

**Comment by derekantrican** [comment]
@jonas0b1011001 ping on this - if you have a chance, I'd like to pull these in and release a new version. Let me know if you want any clarification on my comments

**Comment by jonas0b1011001** [comment]
Done, sorry for the delay.

**Review by derekantrican — APPROVED** [review]
No worries. Thanks for updating it!

**Pull request merged**